Fix manager to not delete and set identical voice commands

renonick87 opened the pull request:
Fixes #433 

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [X] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [X] I have verified that this PR passes lint validation
- [X] I have run the unit tests with this PR
- [X] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).

#### Unit Tests
Added unit tests to verify that VoiceCommands are added/deleted correctly.

### Changelog
##### Bug Fixes
* The VoiceCommandManager will no longer delete a VoiceCommand if the VoiceCommand is going to be immediately added again as part of the same update operation. Below is sample code that can be used to test the functionality. Option 1-3 will be added as part of the first operation. The second operation will delete Option 3 and add Option 4 while leaving Option 1/2 untouched.
```
const voiceCommands = [
    new SDL.manager.screen.utils.VoiceCommand(['Option 1'], () => {}),
    new SDL.manager.screen.utils.VoiceCommand(['Option 2'], () => {}),
    new SDL.manager.screen.utils.VoiceCommand(['Option 3'], () => {}),
];

await screenManager.setVoiceCommands(voiceCommands);
const newVoiceCommands = [
    voiceCommands[0],
    voiceCommands[1],
    new SDL.manager.screen.utils.VoiceCommand(['Option 4'], () => {}),
]
await screenManager.setVoiceCommands(newVoiceCommands);
```

### CLA
- [X] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

crokita commented on lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js:
```diff
@@ -82,18 +82,24 @@ class _VoiceCommandUpdateOperation extends _Task {
         this.onFinished();
     }
 
+    _setOldVoiceCommands (oldVoiceCommands) {
```
This doesn't look like it's used anywhere

crokita reviewed (commented):
See minor feedback

renonick87 commented:
Updates made since last review:
- Equality check on voice commands is not done at the start.
- Added a clone method to the VoiceCommand class and unit tests to cover it.
- The manager's _voiceCommands is set to null if the commands are invalid.
- Updated the docs description for getVoiceCommands since it says that it returns the currently set voice commands but does not actually return _currentVoiceCommands.
- Reduced the logic in _voiceCommandsNotInSecondArray to be much cleaner.
- Fixed the case where updating a Voice Command's listener and calling setVoiceCommands would not update the listener.

crokita commented on lib/js/src/manager/screen/utils/_VoiceCommandUpdateOperation.js:
```diff
@@ -119,6 +135,31 @@ class _VoiceCommandUpdateOperation extends _Task {
+     */
+    _voiceCommandsNotInSecondArray (firstArray = null, secondArray = null) {
+        if (!Array.isArray(firstArray) || firstArray.length === 0) {
+            return secondArray;
```
Elements from the second array should not be in the result according to the docs. Should return `[]` in this case
```suggestion
            return [];
```

crokita reviewed (changes requested):
See feedback
